### PR TITLE
Better support for modules

### PR DESCRIPTION
**Pull request by aspeddro**
**Add support for `module_pack` in parameters**

https://github.com/rescriptbr/rescript-urql/blob/main/src/Client.res#L248-L251

**Support `module_pack` after `as`**

https://github.com/rescriptbr/rescript-urql/blob/main/src/Client.res#L327

**Other changes**

1. Removed `module_pack_type`
2. A new rule to match this cases `repeat1(seq($.module_identifier, '.'))`
3. Two new rules `constrain_module` and `constrain_type`

### CODE REVIEW

**Comment by aspeddro** [comment]
881aae7c2cee0560d56acda72f95ca3a0c6745b9 contains some improvements.

File: grammar.js [hunk 1]
```diff
@@ -1220,54 +1216,57 @@ module.exports = grammar({
...
-      repeat1(seq($.module_identifier, '.')),
+      $.module_identifier_path,
+      '.',
       $.type_identifier
     ),
 
     module_expression: $ => choice(
-      $.module_identifier,
       $.module_identifier_path,
+      $.type_identifier_path,
```
Comment (aspeddro) [inline grammar.js:1226]: `type_identifier_path` to match `module_pack` in `type_declaration`

```res
type t = Ok | Failed(module(Component.t))
//                                    ^ here
```

**Comment by aspeddro** [comment]
The commit 630e76a2b074464524205c9d28c932402ee31560 covers more cases and add more tests.

**Comment by nkrkv** [comment]
Merged. Thanks!

**Pull request merged**